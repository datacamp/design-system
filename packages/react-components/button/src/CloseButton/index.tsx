import { CrossIcon } from '@datacamp/waffles-icons';
import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';
import React from 'react';

const CloseButton: React.FC<{
  className?: string;
  disabled?: boolean;
  onClick: () => void;
  size?: 'small' | 'large';
}> = ({ className, disabled = false, onClick, size = 'large' }) => (
  <button
    aria-label="Close"
    className={className}
    css={css({
      ':active': { color: tokens.color.primary.green.value.rgb },
      ':disabled': { color: '#D1D3D8' },
      ':disabled:hover, :disabled:focus': { backgroundColor: 'transparent' },
      ':focus': {
        backgroundColor: tokens.color.neutral.grey200.value.rgb,
      },
      ':hover': { backgroundColor: tokens.color.neutral.grey200.value.rgb },
      alignItems: 'center',
      backgroundColor: 'white',
      border: 'none',
      borderRadius: 2,
      color: tokens.color.primary.navyText.value.rgb,
      cursor: disabled ? 'not-allowed' : 'pointer',
      display: 'flex',
      flexDirection: 'column',
      height: size === 'large' ? 32 : 20,
      justifyContent: 'center',
      outline: 'none',
      padding: 0,
      width: size === 'large' ? 32 : 20,
      zIndex: 1,
    })}
    disabled={disabled}
    onClick={onClick}
    type="button"
  >
    <CrossIcon aria-hidden size={size === 'large' ? 18 : 12} />
  </button>
);

export default CloseButton;
