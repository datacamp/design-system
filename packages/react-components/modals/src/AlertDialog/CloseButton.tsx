import { CrossIcon } from '@datacamp/waffles-icons';
import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';
import React from 'react';

const CloseButton: React.FC<{ onClick: () => void; disabled: boolean }> = ({
  onClick,
  disabled,
}) => (
  <button
    aria-label="Close"
    css={css({
      ':active': { color: tokens.color.opaque.primary.value.rgb },
      ':disabled': { color: '#D1D3D8' },
      ':disabled:hover, :disabled:focus': { backgroundColor: 'transparent' },
      ':focus': {
        backgroundColor: tokens.color.opaque.porcelain.value.rgb,
      },
      ':hover': { backgroundColor: tokens.color.opaque.porcelain.value.rgb },
      alignItems: 'center',
      backgroundColor: 'white',
      border: 'none',
      borderRadius: 2,
      color: tokens.color.opaque.greyDark.value.rgb,
      cursor: !disabled ? 'pointer' : 'not-allowed',
      display: 'flex',
      flexDirection: 'column',
      height: tokens.size.space[32].value,
      justifyContent: 'center',
      outline: 'none',
      padding: 0,
      position: 'absolute',
      right: tokens.size.space[8].value,
      top: tokens.size.space[8].value,
      width: tokens.size.space[32].value,
    })}
    disabled={disabled}
    onClick={!disabled ? onClick : undefined}
    type="button"
  >
    <CrossIcon size={18} />
  </button>
);

export default CloseButton;
