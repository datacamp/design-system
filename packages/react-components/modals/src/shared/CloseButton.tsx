import { CrossIcon } from '@datacamp/waffles-icons';
import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';
import React from 'react';

const CloseButton: React.FC<{
  className?: string;
  disabled: boolean;
  onClick: () => void;
}> = ({ onClick, disabled = false, className }) => (
  <button
    aria-label="Close"
    className={className}
    css={css({
      ':active': { color: tokens.color.opaque.primary.value.rgb },
      ':disabled': { color: '#D1D3D8' },
      ':disabled:hover, :disabled:focus': { backgroundColor: 'transparent' },
      ':focus': {
        backgroundColor: tokens.color.opaque.porcelain.value.rgb,
      },
      ':hover': { backgroundColor: tokens.color.opaque.porcelain.value.rgb },
      alignItems: 'center',
      backgroundColor: 'white',
      border: 'none',
      borderRadius: 2,
      color: tokens.color.opaque.greyDark.value.rgb,
      cursor: disabled ? 'not-allowed' : 'pointer',
      display: 'flex',
      flexDirection: 'column',
      height: tokens.size.space[32].value,
      justifyContent: 'center',
      outline: 'none',
      padding: 0,
      position: 'absolute',
      right: tokens.size.space[8].value,
      top: tokens.size.space[8].value,
      width: tokens.size.space[32].value,
      zIndex: 1,
    })}
    disabled={disabled}
    onClick={onClick}
    type="button"
  >
    <CrossIcon size={18} />
  </button>
);

export default CloseButton;
