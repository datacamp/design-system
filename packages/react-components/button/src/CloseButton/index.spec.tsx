import '@testing-library/jest-dom/extend-expect';

import axeRender from '@datacamp/waffles-axe-render';
import userEvent from '@testing-library/user-event';
import React from 'react';

import CloseButton from '.';

describe('CloseButton', () => {
  it('renders the close button', async () => {
    const { container } = await axeRender(<CloseButton onClick={() => {}} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('sets className on the button element', async () => {
    const testClass = 'test-className';
    const { container } = await axeRender(
      <CloseButton className={testClass} onClick={() => {}} />,
    );

    expect(container.firstChild).toHaveClass(testClass);
  });

  it('renders the disabled property', async () => {
    const { container } = await axeRender(
      <CloseButton disabled onClick={() => {}} />,
    );

    expect(container.firstChild).toHaveAttribute('disabled');
  });

  it('calls the function on click', async () => {
    const onClick = jest.fn();
    const { container } = await axeRender(<CloseButton onClick={onClick} />);
    userEvent.click(container.firstChild);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
