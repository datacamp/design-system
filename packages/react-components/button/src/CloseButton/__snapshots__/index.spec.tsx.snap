// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders the close button 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border: none;
  border-radius: 2px;
  color: #05192D;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 32px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline: none;
  padding: 0;
  width: 32px;
  z-index: 1;
}

.emotion-0:active {
  color: #03EF62;
}

.emotion-0:disabled {
  color: #E1E1E8;
}

.emotion-0:disabled:hover,
.emotion-0:disabled:focus {
  background-color: transparent;
}

.emotion-0:focus {
  background-color: #EFEFF5;
}

.emotion-0:hover {
  background-color: #EFEFF5;
}

<button
  aria-label="Close"
  class="emotion-0"
  type="button"
>
  <svg
    aria-hidden="true"
    height="16"
    role="img"
    viewBox="0 0 18 18"
    width="16"
  >
    <path
      d="M9.005 7.625l4.83-4.83a.976.976 0 011.38 1.38l-4.83 4.83 4.82 4.82a.976.976 0 11-1.38 1.38l-4.82-4.82-4.83 4.83a.976.976 0 01-1.38-1.38l4.83-4.83-4.84-4.84a.976.976 0 111.38-1.38l4.84 4.84z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</button>
`;
